### DISCUSSION TITLE
Dinamic schedules in adonis

### DISCUSSION THREAD

**Discussion by AndreyViktor** [adonisjs/core#1581] [Help]
hello, I am trying to schedule tasks for each new register in the database, these tasks should run at a specific time that i will receive from the http request.
I tried doing that using events and node-schedule, at the moment I create a new item, it triggers an event and the event is set to the specific time, but it is not working, I mean, it just doesnt run the code when it reaches the time to it.
I also searched for adonis-schedule but cant figure it out how to run this in that way.

here is my code for the event:

const Event = use('Event')
const {parseISO, isBefore, subHours} = require('date-fns')
const schedule = require('node-schedule')

Event.on('new::auction', async (auction)=>{
    const endTime =auction.dt_end

    const j = schedule.scheduleJob(parseISO(endTime), ()=>{
        console.log('end of the auction')
    })
})

any idea on how to create those tasks in adonis?
thanks.

**Comment by prashanth1k** [comment]
Does `parseISO(endTime)` return date object?

Alternatively, there is [adonis-bull](https://github.com/Rocketseat/adonis-bull#scheduled-job) !?
